### ISSUE TITLE
Forward Pass Breaks Autograd and Hardcodes CUDA

### ISSUE THREAD

**Issue by FIRS-ZOFIS** [LianpoWang/SLIIF#35]
In the RESMLPS.forward method, the current implementation converts tensors to NumPy arrays and then back to tensors on a fixed CUDA device:
python

res_list = [mlp(x, cood_out).cpu().detach().numpy() for mlp in self.mlps]
mlp_res = torch.tensor(res_list).cuda()

This approach breaks the autograd graph (since .detach().numpy() discards gradient history) and forces computation onto CUDA, making the model non‑trainable on CPU or other devices. A better practice is to maintain all operations as tensors and preserve the original device:
python

res_list = [mlp(x, cood_out) for mlp in self.mlps]
mlp_res = torch.stack(res_list, dim=0).to(x.device)
